packages/js/components/src/tree-select-control/index.js

```diff
@@ -0,0 +1,497 @@
+ * @property {string} [key] Optional unique key for the Option. It will fallback to the value property if not defined
+ */
+
+/**
+ * @typedef {Object} BaseOption
+ * @property {string}   label      The label for the option
+ * @property {Option[]} [children] The children Option objects
+ *
+ * @typedef {CommonOption & BaseOption} Option
+ */
+
+/**
+ * @typedef {Object} BaseInnerOption
+ * @property {string|JSX.Element}      label          The label string or label with highlighted react element for the option.
+ * @property {InnerOption[]|undefined} children       The children options. The options are filtered if in searching.
+ * @property {boolean}                 hasChildren    Whether this option has children.
+ * @property {InnerOption[]}           leaves         All leaf options that are flattened under this option. The options are filtered if in searching.
+ * @property {boolean}                 checked        Whether this option is checked.
+ * @property {boolean}                 partialChecked Whether this option is partially checked.
+ * @property {boolean}                 expanded       Whether this option is expanded.
+ * @property {boolean}                 parent         The parent of the current option
+ * @typedef {CommonOption & BaseInnerOption} InnerOption
+ */
+
+/**
+ * Renders a component with a searchable control, tags and a tree selector.
+ *
+ * @param {Object}                     props                              Component props.
+ * @param {string}                     [props.id]                         Component id
+ * @param {string}                     [props.label]                      Label for the component
+ * @param {string | false}             [props.selectAllLabel]             Label for the Select All root element. False for disable.
+ * @param {string}                     [props.help]                       Help text under the select input.
+ * @param {string}                     [props.placeholder]                Placeholder for the search control input
+ * @param {string}                     [props.className]                  The class name for this component
+ * @param {boolean}                    [props.disabled]                   Disables the component
+ * @param {Option[]}                   [props.options]                    Options to show in the component
+ * @param {string[]}                   [props.value]                      Selected values
+ * @param {number}                     [props.maxVisibleTags]             The maximum number of tags to show. Undefined, 0 or less than 0 evaluates to "Show All".
+ * @param {Function}                   [props.onChange]                   Callback when the selector changes
+ * @param {(visible: boolean) => void} [props.onDropdownVisibilityChange] Callback when the visibility of the dropdown options is changed.
+ * @return {JSX.Element} The component
+ */
+const TreeSelectControl = ( {
+	id,
+	label,
+	selectAllLabel = __( 'All', 'google-listings-and-ads' ),
```

Comment:
Do we need to change the domain to `woocommerce`?

Reply:
Great catch. Fixed here https://github.com/woocommerce/woocommerce/pull/33384/commits/a13331ad9f15f21134221c7cb35d36b52c22775b